WordPressKit/HTTPRequestBuilder.swift

```diff
@@ -174,30 +178,48 @@ final class HTTPRequestBuilder {
         return request
     }
 
-    func encodeMultipartForm(request: inout URLRequest, forceWriteToFile: Bool) throws -> Either<Data, URL> {
+    func encodeMultipartForm(request: inout URLRequest, forceWriteToFile: Bool) throws -> Either<Data, URL>? {
         guard let multipartForm, !multipartForm.isEmpty else {
-            return .left(Data())
+            return nil
         }
 
         let boundery = String(format: "wordpresskit.%08x", Int.random(in: Int.min..<Int.max))
         request.setValue("multipart/form-data; boundary=\(boundery)", forHTTPHeaderField: "Content-Type")
         return try multipartForm
             .multipartFormDataStream(boundary: boundery, forceWriteToFile: forceWriteToFile)
+    }
+
+    func encodeXMLRPC(request: inout URLRequest, forceWriteToFile: Bool) throws -> Either<Data, URL>? {
+        guard let xmlrpcRequest else {
+            return nil
+        }
 
+        request.setValue("text/xml", forHTTPHeaderField: "Content-Type")
+        let encoder = WPXMLRPCEncoder(method: xmlrpcRequest.method, andParameters: xmlrpcRequest.parameters)
+        if forceWriteToFile {
+            let fileName = "\(ProcessInfo.processInfo.globallyUniqueString)_file.xmlrpc"
+            let fileURL = URL(fileURLWithPath: NSTemporaryDirectory()).appendingPathComponent(fileName)
+            try encoder.encode(toFile: fileURL.path)
+
+            var fileSize: AnyObject?
+            try (fileURL as NSURL).getResourceValue(&fileSize, forKey: .fileSizeKey)
+            if let fileSize = fileSize as? NSNumber {
+                request.setValue(fileSize.stringValue, forHTTPHeaderField: "Content-Length")
+            }
+
+            return .right(fileURL)
+        } else {
+            let data = try encoder.dataEncoded()
+            request.setValue("\(data.count)", forHTTPHeaderField: "Content-Length")
+            return try .left(encoder.dataEncoded())
+        }
```

Comment:
This implementation is the same as the existing one.

https://github.com/wordpress-mobile/WordPressKit-iOS/blob/ce6b99a0d32c2d651a85cf64227aafaef5532fff/WordPressKit/WordPressOrgXMLRPCApi.swift#L232-L255